apis/project_api/project_api/api/scenario.py

```diff
@@ -92,20 +92,25 @@ def download_ens_scenario():
 
     server.app.logger.debug(f"Token parameters {project_id}, {station}, {im_component}")
 
+    project_dir = server.BASE_PROJECTS_DIR / version_str / project_id
     # Load the data
     ensemble_scenario = sc.scenario.EnsembleScenarioResult.load(
-        server.BASE_PROJECTS_DIR
-        / version_str
-        / project_id
-        / "results"
-        / station
-        / im_component
-        / "scenario",
+        project_dir / "results" / station / im_component / "scenario",
     )
-
+    # Load rupture metadata
+    rupture_metadata = utils.load_scenario_rupture_metadata(
+        project_dir,
+        project_id,
+        station,
+        sc.im.IMComponent[im_component],
+        list(
+            sc.scenario.filter_ruptures(ensemble_scenario).to_dict()["mu_data"].keys()
+        ),
+    )
+    # breakpoint()
     with tempfile.TemporaryDirectory() as tmp_dir:
         zip_ffp = au.api.create_scenario_download_zip(
-            ensemble_scenario, tmp_dir, prefix=f"{project_id}",
+            ensemble_scenario, rupture_metadata, tmp_dir
```

Comment:
Does this still need the prefix as a parameter?

Reply:
Yup, we do not use the prefix for now but looks like it will be used at some points as every download function actually does have a prefix param but it's optional.